src/synchronizer.ts

```diff
@@ -26,22 +36,113 @@ export class Synchronizer {
-      this.logger.info(`Assigning issue #${issue.number} to project`);
-      return await this.updateOneIssue(issue, context.config?.projectField);
+      this.logger.debug(`Received event: ${context.eventName}`);
+      if (this.shouldAssignIssue(context.payload, context.config?.labels)) {
+        this.logger.info(`Assigning issue #${issue.number} to project`);
+        return await this.updateOneIssue(issue, context.config?.projectField);
+      } else {
+        return this.logger.info("Skipped assigment as it didn't fullfill requirements.");
+      }
     } else {
       const failMessage = `Event '${context.eventName}' is not expected. Failing.`;
       this.logger.warning(failMessage);
       throw new Error(failMessage);
     }
   }
 
+  /**
+   * Labels can be either an array of objects or an array of string (or maybe both?)
+   * This functions cleans them and returns all the labels names as a string array
+   */
+  convertLabelArray(labels?: (string | { name?: string })[]): string[] {
+    if (!labels || labels.length === 0) {
+      return [];
+    }
+    const list: string[] = [];
+
+    labels.forEach((label) => {
+      if (typeof label === "string" || label instanceof String) {
+        list.push(label as string);
+      } else if (label.name) {
+        list.push(label.name);
+      }
+    });
+
+    return list;
+  }
+
+  /**
+   * Method which takes all of the (predicted) cases and calculates if the issue should be assigned of skipped
+   * @param payload object which contains both the event, the issue type and it's information
+   * @param labels labels required for the action. Can be null or empty
+   * @returns true if the label should be assigned, false if it should be skipped
+   */
+  shouldAssignIssue(payload: Payload, labels?: string[]): boolean {
+    const action = payload.action as IssueEvent;
+
+    if (action === "labeled") {
+      const labelName = payload.label?.name;
+      // Shouldn't happen. Throw and find out what is this kind of event.
+      if (!labelName) {
+        throw new Error("No label found in a labeling event!");
+      }
+
+      this.logger.info(`Label ${labelName} was added to the issue.`);
+
+      // If this is a labeling event but there are no labels in the config we skip them
+      if (!labels || labels.length === 0) {
+        this.logger.notice("No required labels found for event. Skipping assignment.");
+        return false;
+      }
+
+      if (toLowerCase(labels).indexOf(labelName.toLowerCase()) > -1) {
+        this.logger.info(`Found matching label '${labelName}' in required labels.`);
+        return true;
+      } else {
+        this.logger.notice(
+          `Label '${labelName}' does not match any of the labels '${JSON.stringify(labels)}'. Skipping.`,
+        );
+        return false;
+      }
+    } else if (action === "unlabeled") {
+      this.logger.warning("No support for 'unlabeled' event. Skipping");
+      return false;
+    } else {
```

Comment:
`else` is not needed (before the `if` and `else if` returns). Consider removing it to reduce nesting.